examples/pytorch/mnist_trainer.py

```diff
@@ -28,20 +28,53 @@
     from pytorch_lightning.demos.mnist_datamodule import MNISTDataModule
     from pytorch_lightning.plugins.precision import MixedPrecisionPlugin
 
-from lightning_habana import HPUAccelerator, HPUPrecisionPlugin, SingleHPUStrategy
+from lightning_habana import HPUAccelerator, HPUParallelStrategy, HPUPrecisionPlugin, SingleHPUStrategy
+from lightning_habana.utils.resources import device_count
 
-RUN_TYPE = ["basic", "autocast"]
+RUN_TYPE = ["basic", "autocast", "recipe_caching"]
+PLUGINS = ["None", "HPUPrecisionPlugin", "MixedPrecisionPlugin"]
 
 
-def run_trainer(model, plugin):
+def parse_args():
+    """Cmdline arguments parser."""
+    parser = argparse.ArgumentParser(description="Example to showcase mixed precision training with HPU.")
+
+    parser.add_argument("-v", "--verbose", action="store_true", help="Enable verbosity")
+    parser.add_argument(
+        "-r", "--run_types", nargs="+", choices=RUN_TYPE, default=RUN_TYPE, help="Select run type for example"
+    )
+    parser.add_argument(
+        "-p",
+        "--plugins",
+        nargs="+",
+        default=PLUGINS,
+        choices=PLUGINS,
+        help="Plugins for use in training",
+    )
+    return parser.parse_args()
+
+
+def run_trainer(model, plugin, run_type):
     """Run trainer.fit and trainer.test with given parameters."""
     _data_module = MNISTDataModule(batch_size=32)
+    _devices = 1
+    _strategy = SingleHPUStrategy()
```

Comment:
caching for single strategy ?

Reply:
Caching is for sharing recipes compiled on one card with other. Scenario is invalid when training on only one card